File: Warehouse.Server/Model/Entities/Income.cs
namespace Warehouse.Server.Model.Entities
{
  public class Income
  {
    public int Id { get; set; }

    public required string Number { get; set; }
    public DateOnly Date { get; set; }

    public ICollection<IncomeResource> IncomeResources { get; set; } = new List<IncomeResource>();
  }
}

File: warehouse.client/src/app/state/incomes/income.ts

export type IncomeResource = {
  id: number;
  count: number;
  resource: string;
  resourceId: number;
  measure: string;
  measureId: number;
}

export type Income = {
  id: number;
  number: string;
  date: Date;
  items: Array<IncomeResource>;
}
